### PR TITLE
[ADD] Adds rma_claim_checklist

### PR DESCRIPTION
**Pull request by daramousk**

### CODE REVIEW

File: rma_claim_checklist/models/crm_claim.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    def write(self, vals):
+        stage_id = vals.get('stage_id')
+        if stage_id:
+            stage = self.env['crm.claim.stage'].browse(stage_id)
+            for rec in self:
+                for claim_line in rec.claim_line_ids:
+                    states = []
+                    for question in claim_line.question_ids:
+                        states.append(question.checklist_id.claim_state.name)
+                    claim_line.show_questions = True if stage.name in \
```
Comment (rousseldenis) [inline rma_claim_checklist/models/crm_claim.py:20]: @daramousk Why not creating the field 'show_question' as computed field ?

File: rma_claim_checklist/models/crm_claim.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    def _get_questions(self, product):
+        checklist_model = self.env['rma.claim.checklist']
+        questions = self.env['rma.claim.checklist.question']
+        checklists = checklist_model.search(
+            ['|', '|',
+             ('product_id.id', '=', product.id),
+             ('product_brand_id.id', '=', product.brand_id.id),
+             ('product_category_id.id', '=', product.categ_id.id)])
+        for checklist in checklists:
```
Comment (rousseldenis) [inline rma_claim_checklist/models/crm_claim.py:69]:
```suggestion
        return checklists.mapped('question_ids')
```

File: rma_claim_checklist/models/crm_claim.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                vals.update(
+                    {'question_ids':
+                     [(6, 0, questions.ids)],
+                     'show_questions': show_questions})
+        return vals
+
+    def _get_questions(self, product):
+        checklist_model = self.env['rma.claim.checklist']
+        questions = self.env['rma.claim.checklist.question']
+        checklists = checklist_model.search(
```
Comment (rousseldenis) [inline rma_claim_checklist/models/crm_claim.py:65]: Maybe add the domain in a separate function to easy override.

File: rma_claim_checklist/models/crm_claim.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+# Copyright 2017 Therp BV <https://therp.nl>.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, fields, models
+
+
+class CrmClaim(models.Model):
+    _inherit = 'crm.claim'
+
+    @api.multi
+    def write(self, vals):
```
Comment (YogeshMahera-SerpentCS) [inline rma_claim_checklist/models/crm_claim.py:11]: I think we don't need to write a function: we can use the compute function.

**Review by rousseldenis — APPROVED** [review]
Code review. Minor comments.

**Comment by rousseldenis** [comment]
@YogeshMahera-SerpentCS Do you plan to go further in review ?

**Pull request closed without merging**